packages/contract/lib/execute.js

```diff
@@ -95,6 +95,11 @@ const execute = {
     }
     //skipNetworkCheck flag used to skip network call for read data (calls type) methods invocation
     if (skipNetworkCheck) {
+      // Having gasPrice set (often as default gas price in truffle-config) can also cause
+      // view function fail with "insufficient funds for gas" errors"
+      // For calls intending to skip network check, it should also skip this check to.
+      // A workaround would be simply having the gasPrice field delted
+      delete params.gasPrice;
```

Comment:
I have a few comments here:

1. If the `skipNetworkCheck` variable is going to now also perform this function, it should probably be renamed to `isCall` or something.  (Although, if you see my comment (4) below, you'll see that I think that actually this is the wrong place for this code, so there wouldn't be a need to rename the variable after all.)
2. I'm wary of using the `delete` operator here; it's probably fine, but just to be safe, you might be better off writing something like `params = { ...params, gasPrice: 0 }` or similar.  (Or perhaps `undefined` rather than `0` if you want to be even safer.)
3. The `maxFeePerGas` and `maxPriorityFeePerGas` fields can also cause this problem, so they should be treated similarly.
4. I think this isn't the right place for this code, as per [this comment](https://github.com/trufflesuite/truffle/issues/4457#issuecomment-977276979) on the issue.  With the code as it is, it will not only delete any defaults, but also any ones that are manually set, which I don't think we should be doing.  I think instead this code should live in the `getTxParams` function in `utils/index.js`.  Then, you could have an `isCall` flag, and if it were set you could have it ignore certain fields from `constructor.class_defaults`.  So defaults would be ignored, but manually set ones would be respected.

So, I think this is a good start, but not quite the right solution.  But thanks for kicking this off and bringing the issue to our attention!

Reply:
Thank you, please help to find the right place to fix. This pull request can be seen as just a way to showing the problem!

Reply:
OK!  I'm a little unclear here, does that mean you are intending to turn this over to us rather than complete it yourself?  We can do that if you don't have time, but you should say so explicitly if that's your intention... we all already have quite a bit to do, after all, and will take the approach that's less work for us if possible. :)

Reply:
OK, since it's been 29 days, I'm going to take this over if you don't mind... I think I'm going to put up a new PR that supersedes this one, but let me know if you still want to finish this yourself!

Reply:
OK, I've put up PR #4606 to finish this.

Reply:
Hi @haltman-at , thanks for taking it forward. I didn't have time to follow up further since then.

Reply:
No problem!